browser-test/src/applicant/application_navigation.test.ts

```diff
@@ -49,39 +53,299 @@ test.describe('Applicant navigation flow', () => {
+            'program-review',
+            /* fullPage= */ true,
+            /* mobileScreenshot= */ true,
+          )
+        })
+
+        await test.step('can submit application', async () => {
+          await applicantQuestions.submitFromReviewPage()
+          await applicantQuestions.expectConfirmationPage()
+        })
+      })
+
+      test('can skip optional questions', async () => {
+        const {applicantQuestions} = ctx
+        await applicantQuestions.applyProgram(programName)
+
+        await test.step('screen 1', async () => {
+          await applicantQuestions.answerDateQuestion('2021-11-01')
+          await applicantQuestions.answerEmailQuestion('test1@gmail.com')
+        })
+
+        await test.step('screen 2', async () => {
+          await applicantQuestions.clickNext()
+          await applicantQuestions.validateQuestionIsOnPage(staticQuestionText)
+        })
+
+        await test.step('screen 3', async () => {
+          await applicantQuestions.clickNext()
+          await applicantQuestions.validateQuestionIsOnPage(addressQuestionText)
+          await applicantQuestions.answerAddressQuestion(
+            '1234 St',
+            'Unit B',
+            'Sim',
+            'WA',
+            '54321',
+          )
+        })
+
+        await test.step('screen 4 (optional)', async () => {
+          await applicantQuestions.clickNext()
+          await applicantQuestions.validateQuestionIsOnPage(radioQuestionText)
+          // Don't answer the optional radio question
+        })
+
+        await test.step('screen 5 (currency optional)', async () => {
+          await applicantQuestions.clickNext()
+          await applicantQuestions.validateQuestionIsOnPage(phoneQuestionText)
+          await applicantQuestions.validateQuestionIsOnPage(
+            currencyQuestionText,
+          )
+
+          await applicantQuestions.answerPhoneQuestion('4256373270')
+          // Don't answer the optional currency question
+        })
+
+        await test.step('can submit application', async () => {
+          await applicantQuestions.clickNext()
+          await applicantQuestions.expectReviewPage()
+          await applicantQuestions.submitFromReviewPage()
+          await applicantQuestions.expectConfirmationPage()
+        })
+      })
+
+      test('answering questions out of order', async () => {
+        const {page, applicantQuestions} = ctx
+        await enableFeatureFlag(page, 'save_on_all_actions')
+        await logout(page)
+
+        await applicantQuestions.clickApplyProgramButton(programName)
+
+        await test.step('answer screen 4', async () => {
+          await applicantQuestions.answerQuestionFromReviewPage(
+            radioQuestionText,
+          )
+          await applicantQuestions.validateQuestionIsOnPage(radioQuestionText)
+          await applicantQuestions.answerRadioButtonQuestion('one')
+          await applicantQuestions.clickReview()
+
+          await validateScreenshot(
```

Comment:
suggestion (non-blocking):  A screenshot seems unnecessary here.  Instead, you could check whether that question has an "Edit" link, no?

Reply:
I think this screenshot is checking that the review page can render some questions as unanswered and other questions as answered? We could definitely check that some questions have "Answer" while some questions have "Edit" instead of using a screenshot if that's better!

(Note that this screenshot existed before this PR, just as `third-question-answered` instead of `fourth-question-answered`.)

Reply:
My preference is to have less screenshots, but since this was already here, I'll leave it up to you on whether you want to change it or not.

Reply:
Agree with the preference to have less screenshots! I left it as-is for now since this PR was focused on adding new tests for optional questions and the "Next" button, but we can remove the screenshot in the future if it's getting annoying to constantly update.